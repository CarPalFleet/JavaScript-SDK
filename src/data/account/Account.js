/**
 * @fileoverview This file contains all account related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import { apiResponseErrorHandler } from '../utility/Util';

/**
 * Request reset password token
 * @param {string} email
 * @return {object} Promise resolve/reject
 */
export const resetPasswordRequestAsync = async (email) => {
  try {
    const response = await axios({
      method: 'POST',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        email,
      },
    });
    return camelize(response);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Reset password with valid token
 * @param {string} token
 * @param {string} email
 * @param {string} password
 * @param {string} passwordConfirmation
 * @return {object} Promise resolve/reject
 */
export const resetPasswordAsync = async (
  token,
  email,
  password,
  passwordConfirmation
) => {
  try {
    const response = await axios({
      method: 'PUT',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
        email,
        password,
        passwordConfirmation,
      },
    });
    return camelize(response.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Validate Reset Password Token
 * @param {string} token
 * @return {object} Promise resolve/reject
 */
export const validateResetPasswordTokenAsync = async (token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: endpoints.API_V3.VALIDATE_PASSWORD_RESET_TOKEN.replace(
        '{token}',
        token
      ),
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
      },
    });
    return camelize(response.data.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};
