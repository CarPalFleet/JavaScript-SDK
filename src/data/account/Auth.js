import axios from 'axios';
import endpoints, { cid, secret } from '../Endpoint';

export const getTokenAsync = async (email, password)=>{
    try{
        const response = await axios({method: 'post', 
                                      url: endpoints.OAUTH,
                                      headers: {'Content-Type': 'application/json'},
                                      data: {
                                        username: email,
                                        password,
                                        grant_type: "password",
                                        client_id: cid,
                                        client_secret: secret,
                                        scope: ""
                                      }})
        return response.data.data;     
    }catch(e){
        return null
    }
}            