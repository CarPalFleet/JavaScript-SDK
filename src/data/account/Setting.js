/**
 * @fileoverview This file contains all Settings related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import {apiResponseErrorHandler} from '../utility/Util';

/** Retrieve User's settings
 * There are 3 setting types in the setting table
 * 1. routing, 2. my-order, 3. driver-list
 * In routing type, it includes user time line setting (15 min, 30 min, 45 min etc.)
 * Retrieve table settings from my-order type OR driver-list.
 * @param {integer} userId
 * @param {string} type # routing, my-order, driver-list
 * @param {string} token
 * @return {object} promise (resolve/reject)
 */
export const getUserSettingsAsync = async (userId, type, token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: `${endpoints.USER_SETTINGS.replace(
        '{0}',
        userId
      )}?type=${type}`,
      headers: {Authorization: token},
    });
    return camelize(response.data);
  } catch (e) {

    return apiResponseErrorHandler(e);
  }
};
