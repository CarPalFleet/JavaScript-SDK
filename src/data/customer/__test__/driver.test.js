import CONFIG from './Config';
import { getTokenAsync } from '../../account/Auth';
import {
  createNewDriverAsync,
  getCustomerDriverDetailAsync,
  getCustomerDriverListAsync,
  getCustomerDriversAsync,
  updateDriverLiveData
} from '../Driver';

test('Creating new driver account by a customer account', async () =>{
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
    const result = getTokenAsync(CONFIG.temail, CONFIG.tpassword, CONFIG.clientId, CONFIG.clientSecret);
    const token = await result;

    const driver = {
        identityId: 1,
        productTypeId: 3,
        transactionGroupId: [180],
        isNewUser: true,
        firstName: 'User',
        lastName: makeid(10),
        email: `${makeid(10)}@example.com`,
        password: '123456',
        birthday: '1980-01-01',
        phone: '+6592341092'
    }

    const response = await createNewDriverAsync(driver, 1, token.accessToken);

    expect('driver' in response).toBe(true);
})

test(`Test for retrieving detail of customer's driver`, async () => {
    const result = getTokenAsync(CONFIG.temail, CONFIG.tpassword, CONFIG.clientId, CONFIG.clientSecret);
    const token = await result;
    const response = await getCustomerDriverDetailAsync(1, 5, 9869, token.accessToken);
    expect(response instanceof Object).toBe(true);
})

test('Test for retrieving drivers by a customer account', async () =>{
    // jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
    // const filterObj = {
    //     driverStatusIds: [2],
    //     orderRouteTypeIds: [1,2],
    //     driverTypeIds: [1,2,3]
    // }
    // const result = getTokenAsync(CONFIG.email, CONFIG.password, CONFIG.clientId, CONFIG.clientSecret);
    // const token = await result;
    //
    // const response = await getCustomerDriverListAsync(filterObj, 1, token.accessToken);
    //
    // expect(response instanceof Array).toBe(true);
    expect(true).toBe(true);
})

test('Test for pubsub live data for job', async () =>{
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
    const originalDriverDatum = {
      "activeStatusCounts":{"1":0,"2":0,"3":0,"4":0},
      "driverTypeCounts":{"1": 1,"2": 3,"3": 4},
      "data":{
         "1":[],
         "2":[],
         "3":[{
           "addressId": 0,
           "driverId": 9168,
           "customerId": 10919,
           "orderId": 62411,
           "driverStatusId": 3,
           "driverTypeIds":[
              1
           ],
           "id":"9b4cf220-cd94-4a0b-84ac-32f74dfc142f",
           "latitude": "1.2788882",
           "longitude": "103.8482516",
           "orderRouteTypeId": 1,
           "updatedAt": "2018-01-11 06:06:57"
         }],
         "4":[]
      }, "totalStatusCounts":0
    }
    const pubSubPayload = {
      data: {
        "addressId": 0,
        "customerId": 10919,
        "driverId": 9168,
        "driverTypeIds":[
          1
        ],
        "id":"1ada3ace-67ab-4e3b-a1e0-a0d3b63fedc8",
        "latitude": "1.2789042",
        "longitude": "103.8482397",
        "orderId": 62411,
        "orderRouteTypeId": 1,
        "updatedAt": "2018-01-11 06:06:14",
        // "driverStatusId": 4
      },
      lastDriverStatusId: 1
  }

    const filterObject = {
      // driverStatusIds: [2],
      orderRouteTypeIds: 1,
      driverTypeIds: [1]
    }
    const result = getTokenAsync(CONFIG.email, CONFIG.password, CONFIG.clientId, CONFIG.clientSecret);
    const token = await result;
    const response = updateDriverLiveData(originalDriverDatum, pubSubPayload, filterObject);
    expect(response instanceof Object).toBe(true);
})

function makeid(size) {
    var text = "";
    var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

    for (var i = 0; i < size; i++)
      text += possible.charAt(Math.floor(Math.random() * possible.length));

    return text;
}
