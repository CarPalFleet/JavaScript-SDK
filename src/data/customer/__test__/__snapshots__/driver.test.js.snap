// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test getDriverCountsAsync should get getDriverCountsAsync success response 1`] = `
Object {
  "activeStatusCounts": Object {
    "1": 0,
    "2": 0,
    "3": 0,
    "4": 0,
  },
  "driverTypeCounts": Object {
    "1": 0,
    "2": 0,
    "3": 0,
  },
  "totalStatusCounts": 0,
}
`;

exports[`Test getDriversWithFiltersAsync should get getDriversWithFiltersAsync success response 1`] = `
Object {
  "data": Object {
    "1": Array [],
    "2": Array [],
    "3": Array [],
    "4": Array [],
  },
}
`;
